src/sentry/search/snuba/backend.py

```diff
@@ -124,150 +166,40 @@ def apply(self, queryset, search_filter):
         return qs_method(**q_dict)
 
 
-def assigned_to_filter(actor, projects):
-    from sentry.models import OrganizationMember, OrganizationMemberTeam, Team
-
-    if isinstance(actor, Team):
-        return Q(assignee_set__team=actor)
-
-    teams = Team.objects.filter(
-        id__in=OrganizationMemberTeam.objects.filter(
-            organizationmember__in=OrganizationMember.objects.filter(
-                user=actor, organization_id=projects[0].organization_id
-            ),
-            is_active=True,
-        ).values("team")
-    )
-
-    return Q(
-        Q(assignee_set__user=actor, assignee_set__project__in=projects)
-        | Q(assignee_set__team__in=teams)
-    )
-
-
-def unassigned_filter(unassigned, projects):
-    from sentry.models.groupassignee import GroupAssignee
-
-    query = Q(
-        id__in=GroupAssignee.objects.filter(project_id__in=[p.id for p in projects]).values_list(
-            "group_id", flat=True
-        )
-    )
-    if unassigned:
-        query = ~query
-    return query
-
+class QuerySetBuilder(object):
+    def __init__(self, conditions):
+        self.conditions = conditions
 
-def get_search_filter(search_filters, name, operator):
-    """
-    Finds the value of a search filter with the passed name and operator. If
-    multiple values are found, returns the most restrictive value
-    :param search_filters: collection of `SearchFilter` objects
-    :param name: Name of the field to find
-    :param operator: '<' or '>'
-    :return: The value of the field if found, else None
-    """
-    assert operator in ("<", ">")
-    comparator = max if operator.startswith(">") else min
-    found_val = None
-    for search_filter in search_filters:
-        # Note that we check operator with `startswith` here so that we handle
-        # <, <=, >, >=
-        if search_filter.key.name == name and search_filter.operator.startswith(operator):
-            val = search_filter.value.raw_value
-            found_val = comparator(val, found_val) if found_val else val
-    return found_val
+    def build(self, queryset, search_filters):
+        for search_filter in search_filters:
+            name = search_filter.key.name
+            if name in self.conditions:
+                condition = self.conditions[name]
+                queryset = condition.apply(queryset, search_filter)
+        return queryset
 
 
-class SnubaSearchBackend(SearchBackend):
+class AbstractQueryExecutor:
     def query(
         self,
         projects,
-        environments=None,
-        sort_by="date",
-        limit=100,
-        cursor=None,
-        count_hits=False,
-        paginator_options=None,
-        search_filters=None,
-        date_from=None,
-        date_to=None,
+        retention_window_start,
+        group_queryset,
+        environments,
+        sort_by,
+        limit,
+        cursor,
+        count_hits,
+        paginator_options,
+        search_filters,
+        date_from,
+        date_to,
```

Comment:
This is for later: when you will introduce the two implementaiton we will probably try to move some of these parameters into the construction of the object.